src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1643,16 +1659,11 @@ ss::future<ntp_archiver_upload_result> ntp_archiver::upload_segment(
     // segment, so it is okay to ignore the index upload failure, we still
     // want to advance the offsets because the segment did get uploaded.
 
-    // Note: this operation can be started in the background.
-    // In order to do this the context should be associated with the
-    // background operation. We can't background it as is because the
-    // 'upload_index' call is taking 'rtc' as a reference. So there should be
-    // some wrapper for this call.
-    std::ignore = co_await _remote.upload_index(
-      _conf->bucket_name,
-      cloud_storage_clients::object_key{index_path},
-      index,
-      rtc);
+    ssx::spawn_with_gate(
+      _gate,
+      [this, path = std::move(index_path), index = std::move(index)]() mutable {
+          return upload_index(std::move(path), std::move(index));
+      });
```

Comment:
is there any risk here that we accumulate a bunch of pending upload background fibers?

Reply:
sorry @nvartolomei i missed your similar comment.

yeh i don't think i have enough context to evaluate this fully, but the pattern of backgrounding there seemed like it could be risky without other controls (explicit or implicit) that'd keep things in check.

Reply:
I don't think it's a problem. The background fiber uploads an index. In order to start the upload it has to acquire the cloud storage client from the pool first. If we stuck waiting for the cloud storage client because the pool is saturated any `ntp_archiver` instance will have to wait too because the pool is shared. You can think about the pool as the semaphore that limits number of concurrent segment uploads. The background index upload is either waiting for units (and in this case no new segment upload is possible because it will also stuck waiting for units) or it's progressing and there is no risk in this case.

Reply:
oh of course. i was thinking we would keep spawning background fibers at the upload request rate even with the client pool saturated, but yeah that would affect the upload loop in the foreground as well.

Reply:
yeah i don't disagree. i'd like to add something more explicit (maybe even for backport), but my read is that it's not a major concern for the release, assuming nothing else comes up.